src/PointwiseFunctions/AnalyticData/GrMhd/InitialMagneticFields/Toroidal.hpp

```diff
@@ -0,0 +1,115 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <limits>
+
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Options/Options.hpp"
+#include "PointwiseFunctions/AnalyticData/GrMhd/InitialMagneticFields/InitialMagneticField.hpp"
+#include "PointwiseFunctions/GeneralRelativity/TagsDeclarations.hpp"
+#include "PointwiseFunctions/Hydro/TagsDeclarations.hpp"
+#include "Utilities/TMPL.hpp"
+#include "Utilities/TaggedTuple.hpp"
+
+namespace grmhd::AnalyticData::InitialMagneticFields {
+
+/*!
+ * \brief %Toroidal magnetic field for GRMHD initial data.
+ *
+ * The vector potential has the form
+ *
+ * \f{align*}{
+ *  A_x & = A_y = 0 , \\
+ *  A_z & = A_b \varpi^2 \max(p-p_{\mathrm{cut}}, 0)^{n_s} ,
+ * \f}
+ *
+ * where \f$A_b\f$ controls the amplitude of the magnetic field,
+ * \f$\varpi^2=x^2+y^2=r^2-z^2\f$ is the cylindrical radius,
+ * \f$n_s\f$ controls the degree of differentiability, and
+ * \f$p_{\mathrm{cut}}\f$ controls the pressure cutoff below which the magnetic
+ * field is zero.
+ *
+ * On the region where the field is non-zero, the magnetic field is given by:
+ *
+ * \f{align*}{
+ *   B^x & = \frac{A_c}{\sqrt{\gamma}} \left[
+ *           2y(p-p_{\mathrm{cut}})^{n_s}
+ *           + \varpi^2 n_s (p-p_{\mathrm{cut}})^{n_s-1} \partial_y p \right],\\
+ *   B^y & = -\frac{A_c}{\sqrt{\gamma}} \left[
+ *           2x(p-p_{\mathrm{cut}})^{n_s}
+ *           + \varpi^2 n_s (p-p_{\mathrm{cut}})^{n_s-1} \partial_x p \right],\\
+ *   B^z & = 0 .
+ * \f}
```

Comment:
Can you add here and in the `Poloidal` what magnetic field values you have for different `A_b` and other params? It's very annoying to have to figure this out, which is why I documented it for the `MagnetizedTovStar`. I think all that matters is the units, G=c=M_sun=1. See https://spectre-code.org/classgrmhd_1_1AnalyticData_1_1MagnetizedTovStar.html

Reply:
Since evaluations of `Poloidal` and `Toroidal` fields require information from hydro side (pressure and d_pressure), it cannot be documented here. I'll add that table for different `n_s` and `A_b`s in the docs of `MagnetizedTovStar`. Does that sound good?

Reply:
Yep! Thanks!